src/devices/bus/msx/cart/nomapper.cpp

```diff
@@ -50,60 +50,81 @@ image_init_result msx_cart_nomapper_device::initialize_cartridge(std::string &me
 	const u32 size = cart_rom_region()->bytes();
 	u8 *rom = cart_rom_region()->base();
 
-	// determine start address, default to 0x4000
-	m_start_address = 0x4000;
-
-	switch (size)
+	// 64KB images fill all pages.
+	if (size == 0x10000)
+	{
+		m_start_address = 0;
+	}
+	else if (is_loaded_through_softlist())
 	{
-		// 8KB/16KB
-		case 0x2000: case 0x4000:
+		const char *start_page_str = get_feature("start_page");
+		if (!start_page_str)
 		{
-			uint16_t start = rom[3] << 8 | rom[2];
+			message = "msx_cart_nomapper_device: Feature 'start_page' was not found.";
+			return image_init_result::FAIL;
+		}
```

Comment:
Should it always fail here, or would it be better to use the most common starting page if the attribute isn’t present?  If you think it should always be specified explicitly, that’s fair enough.

Reply:
The most common starting address seems to be 0x4000, page 1. If we really wanted to we could use that. I would prefer to have it always explicitly defined in the software list entries so you don't have to remember that there is a special default.
Now it would be either the software item defines which mapper should be used, or the software occupies the full address (for 64kb dumps), or the location in memory is specified through that feature.

I kept in the old determination of the starting_address code only for the cases when non-softlisted software is loaded.